src/EventListener/EventRegistrationFormListener.php

```diff
@@ -52,9 +52,19 @@ public function __invoke(array &$submittedData, array $formData, ?array $files,
 
         $member = $this->getMember();
 
-        $registration = new EventRegistrationModel();
+        // check if registration exists
+        $registration = EventRegistrationModel::findOneBy(['pid = ?', 'member = ?'], [(int) $event->id, (int) $member->id]);
+
+        if (null !== $registration) {
+            $registration->cancelled = 0;
```

Comment:
```suggestion
            $registration->cancelled = 0;
            $registration->confirmed = 0;
```

I think you also need to set `confirmed` to false as the registration would potentially need to be confirmed again.

Reply:
Correct!